corded/objects/gateway.py

```diff
@@ -24,3 +24,90 @@ def typed_data(self):
     @property
     def dispatch_name(self):
         return (self.t or f"op_{self.op}").lower()
+
+
+class Intents:
+    """Represents a discord intents flag
+    Allows for easier passing of intents to the gateway
+    
+    """
+    value: int = 0
+    valid: Dict[str, int] = {
+        "guilds": 1 << 0,
+        "guild_members": 1 << 1,
+        "guild_bans": 1 << 2,
+        "guild_emojis": 1 << 3,
+        "guild_integrations": 1 << 4,
+        "guild_webhooks": 1 << 5,
+        "guild_invites": 1 << 6,
+        "guild_voice_states": 1 << 7,
+        "guild_presences": 1 << 8,
+        "guild_messages": 1 << 9,
+        "guild_message_reactions": 1 << 10,
+        "guild_message_typing": 1 << 11,
+        "direct_messages": 1 << 12,
+        "direct_message_reactions": 1 << 13,
+        "direct_message_typing": 1 << 14
+    }
+    
+    def __init__(self):
+        for flag in self.valid.keys():
+            super(Intents, self).__setattr__(flag, False)
+            
+    def __setattr__(self, name: str, value: bool) -> None:
+        if name in self.valid.keys():
+            if value is True:
+                super(Intents, self).__setattr__("value", self.value + self.valid[name])
+                super(Intents, self).__setattr__(name, value)
+            elif value is False and getattr(self, name) is True:
+                super(Intents, self).__setattr__("value", self.value - self.valid[name])
+                super(Intents, self).__setattr__(name, value)
+        else:
+            raise Exception(f"{name}, is not a valid intent")
+        
+    def __iter__(self):
+        for key, value in self.__dict__.items():
+            if key == "value":
+                continue
+            yield (key, value)
+
+    def __hash__(self):
+        return hash(self.value)
+
+    def __repr__(self):
+        return f"<{self.__class__.__name__} value={self.value}>"
+    
+    def __eq__(self, other):
+        return isinstance(other, self.__class__) and self.value == other.value
+    
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    @classmethod
+    def all(cls):
+        """A classmethod that will enable all intents including privileged ones
+
+        Returns:
+            Intents: The Intents instance that was created
+        """
+        intents = cls.__new__(cls)
+        for flag in cls.valid.keys():
+            cls.__setattr__(intents, flag, True)
+            
+        return intents
+
+    @classmethod
+    def default(cls):
+        """A classmethod that will enable all intents excluding privileged ones
+
+        Returns:
+            Intents: The Intents instance that was created
+        """
+        intents = cls.__new__(cls)
+        for flag in cls.valid.keys():
+            cls.__setattr__(intents, flag, True)
```

Comment:
Same as above comment.

Reply:
I need to do it like this since `intents.__setattr__` will not work